### DISCUSSION TITLE
Hydrating `enumType` as `getArrayResult` gives *string* instead of Enum

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [doctrine/doctrine2#9588] [General]
@derrabus When doing
```php
$queryBuilder = $this->createQueryBuilder('f')
    ->select('f.id, f.suit')
    ->getQuery()
    ->getArrayResult() // Query::HYDRATE_ARRAY
;
```
... with `suit` being an `enumType`, the resulting `suit` column in the array is the *string* value of the Enum's case already (not the case itself).

This looks inconsistent to me: Usually, if I switch from `->getResult()` to `->getArrayResult()`, I don't have to change anything in the template. But in this case, I have to change `{{ f.suit.value }}` to `{{ f.suit }}`.

If you're telling me that it's going to stay this way, I'll document it at https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/reference/basic-mapping.html

**Comment by derrabus** [comment]
Please don't @ people. FWIW, this looks like a bug to me.

**Reply by janedbal** [reply]
Reported as bug: https://github.com/doctrine/orm/issues/9622

**Comment by michnovka** [comment]
I would really like to offer some help on this one, but some internals of Doctrine are still unknown to me. Where should we lead discussion and questions - here, or in the issue? I might have few "stupid" questions, but if you have patience with me, you might get a better contributor from me that way. If this is out of your time capabilities and you wish to fix this yourselves, then I understand. Please let me know

**Reply by greg0ire** [reply]
@michnovka the issue is the best way, discussions are a Stack Overflow like feature
